libs/langgraph/langgraph/prebuilt/chat_agent_executor.py

```diff
@@ -49,89 +49,110 @@ class AgentState(TypedDict):
+        model_input_preprocessor_runnable = RunnableLambda(
             lambda state: [_system_message] + state["messages"],
-            name=STATE_MODIFIER_RUNNABLE_NAME,
+            name=MODEL_INPUT_PREPROCESSOR_RUNNABLE_NAME,
         )
-    elif isinstance(state_modifier, SystemMessage):
-        state_modifier_runnable = RunnableLambda(
-            lambda state: [state_modifier] + state["messages"],
-            name=STATE_MODIFIER_RUNNABLE_NAME,
+    elif isinstance(model_input_preprocessor, SystemMessage):
+        model_input_preprocessor_runnable = RunnableLambda(
+            lambda state: [model_input_preprocessor] + state["messages"],
+            name=MODEL_INPUT_PREPROCESSOR_RUNNABLE_NAME,
         )
-    elif callable(state_modifier):
-        state_modifier_runnable = RunnableLambda(
-            state_modifier, name=STATE_MODIFIER_RUNNABLE_NAME
+    elif callable(model_input_preprocessor):
+        model_input_preprocessor_runnable = RunnableLambda(
+            model_input_preprocessor, name=MODEL_INPUT_PREPROCESSOR_RUNNABLE_NAME
         )
-    elif isinstance(state_modifier, Runnable):
-        state_modifier_runnable = state_modifier
+    elif isinstance(model_input_preprocessor, Runnable):
+        model_input_preprocessor_runnable = model_input_preprocessor
     else:
         raise ValueError(
-            f"Got unexpected type for `state_modifier`: {type(state_modifier)}"
+            f"Got unexpected type for `model_input_preprocessor`: {type(model_input_preprocessor)}"
         )
 
-    return state_modifier_runnable
+    return model_input_preprocessor_runnable
 
 
-def _convert_messages_modifier_to_state_modifier(
+def _convert_messages_modifier_to_model_input_preprocessor(
     messages_modifier: MessagesModifier,
-) -> StateModifier:
-    state_modifier: StateModifier
+) -> ModelInputPreprocessor:
+    model_input_preprocessor: ModelInputPreprocessor
     if isinstance(messages_modifier, (str, SystemMessage)):
         return messages_modifier
     elif callable(messages_modifier):
 
-        def state_modifier(state: AgentState) -> Sequence[BaseMessage]:
+        def model_input_preprocessor(state: AgentState) -> Sequence[BaseMessage]:
             return messages_modifier(state["messages"])
 
-        return state_modifier
+        return model_input_preprocessor
     elif isinstance(messages_modifier, Runnable):
-        state_modifier = (lambda state: state["messages"]) | messages_modifier
-        return state_modifier
+        model_input_preprocessor = (lambda state: state["messages"]) | messages_modifier
+        return model_input_preprocessor
     raise ValueError(
         f"Got unexpected type for `messages_modifier`: {type(messages_modifier)}"
     )
 
 
-def _get_model_preprocessing_runnable(
-    state_modifier: Optional[StateModifier],
+def _get_model_input_preprocessor(
+    preprocess_model_inputs: Optional[ModelInputPreprocessor],
+    state_modifier: Optional[ModelInputPreprocessor],
```

Comment:
why do we need both (for a private method)

Reply:
we don't necessarily, but putting the logic for picking the preprocessor into a single private function keeps the actual `create_react_agent` a bit cleaner imo